pmlc/dialect/tile/transforms/padding.cc

```diff
@@ -223,9 +231,92 @@ void PadPass::runOnFunction() {
     op->setAttr("padLower", builder.getI64ArrayAttr(info.lower));
     op->setAttr("padUpper", builder.getI64ArrayAttr(info.upper));
   }
+
+  // Finally, remove unnecessary constraints
+  func.walk([&](ContractionOp op) {
+    // Early exit if no constraints
+    if (!op.cons())
+      return;
+    auto numDims = op.sink().getNumDims();
+    // Otherwise, we will check each constraint against each input
+    llvm::SmallVector<AffineExpr, 8> savedConstraints;
+    llvm::SmallVector<llvm::SmallVector<AffineExpr, 10>, 2> inputs;
+    // Go over all the input tensors + makey polyhedra for any padded ones
+    for (size_t i = 0; i < op.getNumTensors(); i++) {
```

Comment:
FYI, if we try to get this upstream this will be marked for a change. The coding standards (https://llvm.org/docs/CodingStandards.html) dictate that for loops should not recompute the end condition and always use prefix increments (not postfix).

Reply:
Specifically https://llvm.org/docs/CodingStandards.html#don-t-evaluate-end-every-time-through-a-loop